File: fortran/problem_2.f90
program  problem_2
    integer a
    integer b
    integer c
    integer total
    a = 1
    b = 1
    c = 0
    total = 0
    do while (c < 4000000)
        c = a + b
        a = b
        b = c

        if (MODULO(c, 2) == 0) then
            total = total + c
        end if
    end do
    print *,"Problem 2:", total
end program problem_2

File: c/problem_2.c
#include <stdio.h>

int main() 
{
    unsigned long int total; // It's a really large number...
    int a = 1;
    int b = 1;
    int c = 0;
    while(c < 4000000) {
        c = a + b;
        a = b;
        b = c;
        if (c % 2 == 0) {
            total += c;
        }
    }
    printf("Problem 2: %lu\n", total);
    return 0;
}